community/kernel/src/main/java/org/neo4j/kernel/impl/store/id/IdGeneratorImpl.java

```diff
@@ -64,31 +60,15 @@
  */
 public class IdGeneratorImpl implements IdGenerator
 {
-    // sticky(byte), nextFreeId(long)
-    public static final int HEADER_SIZE = 9;
-
-    // if sticky the id generator wasn't closed properly so it has to be
-    // rebuilt (go through the node, relationship, property, rel type etc files)
-    private static final byte CLEAN_GENERATOR = (byte) 0;
-    private static final byte STICKY_GENERATOR = (byte) 1;
-
     /**
      * Invalid and reserved id value. Represents special values, f.ex. the end of a relationships/property chain.
      * Please use {@link IdValidator} to validate generated ids.
      */
     public static final long INTEGER_MINUS_ONE = 0xFFFFFFFFL;  // 4294967295L;
 
-    // number of defragged ids to grab from file in batch (also used for write)
-    private int grabSize = -1;
-    private final AtomicLong highId = new AtomicLong( -1 );
-
-    private final File file;
-    private final FileSystemAbstraction fs;
-    private StoreChannel fileChannel;
-
     private final long max;
-    private final boolean aggressiveReuse;
-    private FreeIdKeeper keeper;
+    private final IdContainer idContainer;
+    private long highId = INTEGER_MINUS_ONE;
```

Comment:
No need to initialise it to this strange constant because the constructor will overwrite it.